app/classifier/drawing-tools/components/Fan/fan.jsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import DeleteButton from '../../delete-button';
+import Draggable from '../../../../lib/draggable';
+import DragHandle from '../../drag-handle';
+import DrawingToolRoot from '../../root';
+import deleteIfOutOfBounds from '../../delete-if-out-of-bounds';
+
+const MINIMUM_SIZE = 10;
+
+class Fan extends React.Component {
+
+  static defaultValues({ x, y }) {
+    return {
+      x,
+      y,
+      radius: 0,
+      rotation: 0,
+      spread: 30
+    };
+  }
+
+  static initStart({ x, y }) {
+    return Object.assign({}, Fan.defaultValues({ x, y }), { _inProgress: true });
+  }
+
+  static initMove(cursor, mark) {
+    const radius = DrawingToolRoot.distance(mark.x, mark.y, cursor.x, cursor.y);
+    const rotation = Fan.getCursorAngle(cursor, mark);
+    return Object.assign({}, mark, { radius, rotation });
+  }
+
+  static initRelease() {
+    return { _inProgress: false };
+  }
+
+  static initValid(mark) {
+    return mark.radius > MINIMUM_SIZE;
+  }
+
+  static getCursorAngle(cursor, mark) {
+    // calculates the angle between a cursor position and the mark position.
+    const deltaX = cursor.x - mark.x;
+    const deltaY = cursor.y - mark.y;
+    return Math.atan2(deltaY, deltaX) * (180 / Math.PI);
+  }
+
+  handleDrag(e, d) {
+    const { mark } = this.props;
+    const difference = this.props.normalizeDifference(e, d);
+    mark.x += difference.x;
+    mark.y += difference.y;
+    this.props.onChange(mark);
+  }
+
+  handleRotate(e) {
+    const { mark, getEventOffset } = this.props;
+    const { x, y } = getEventOffset(e);
+    const radius = DrawingToolRoot.distance(mark.x, mark.y, x, y);
+    const rotation = Fan.getCursorAngle({ x, y }, mark);
+    const newMark = Object.assign({}, mark, { radius, rotation });
+    this.props.onChange(newMark);
+  }
+
+  handleSpread(e) {
+    const { mark, getEventOffset } = this.props;
+    const { x, y } = getEventOffset(e);
+    const cursorAngle = Fan.getCursorAngle({ x, y }, mark);
+    const spread = 2 * Math.abs(cursorAngle - mark.rotation);
```

Comment:
The angle difference here needs to take into account when `cursorAngle` passes from `180` to `-180`.  This can be done with a function similar to:

```js
let spread = Math.abs(cursorAngle - mark.rotation);
if (spread > 180) {
    spread = 360 - spread;
}
spread *= 2;
```

Reply:
Good point. I've started writing tests, so I'll add a test for this, then fix the code to pass the test.

Reply:
I've added some tests but they don't trip this bug. 🤔 

I've also added a failing test for a maximum spread angle of 180º.